### PR TITLE
First Missing Positive Number - C

### PR DESCRIPTION
**Pull request by SangeetaMishra**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

First Missing Positive Number in C

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#855

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@SangeetaMishra move this program to `cp` section

File: C/README.md [hunk 1]
```diff
@@ -65,4 +68,5 @@ _add list here_
 ## Other
 - [Number_of_duplicates_using_Binary_search](other/Number_of_duplicates_using_Binary_search.c)
 - [Tower of Hanoi](other/Tower_Of_Hanoi.c)
+- [First Missing Positive Integer](other/first_missing_positive_integer.c)
```
Comment (atarax665) [inline C/README.md]:
```suggestion

```

File: C/cp/first_missing_positive_integer.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#include <stdio.h>
+
+int main(){
+    int n;
+    printf("Enter the value of n: ");
+    scanf("%d\n",&n);
+    printf("Enter the array:\n");
+    int a[n+1];
+    for (int i = 1; i <= n; ++i)
```
Comment (Abhijit2505) [inline C/cp/first_missing_positive_integer.c:10]: Try dynamic memory allocation in C for array.

File: C/cp/first_missing_positive_integer.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        scanf("%d", &a[i]);
+    }
+
+    for (int i = 1; i <= n; i++) {
+        if (a[i] <= 0 || a[i] > n) 
+            continue; 
+
+        int x = a[i]; 
+        while (a[x] != x) {
```
Comment (Abhijit2505) [inline C/cp/first_missing_positive_integer.c]: Give good names to the variables

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]
Small changes required, other than that, It's nice

**Review by iamrajiv — APPROVED** [review]

**Review by Abhijit2505 — APPROVED** [review]

**Comment by Abhijit2505** [comment]
@SangeetaMishra , as you can see there is a merge conflict. Pull from the master repo to your forked repo and you are good to go.
Also, remember the golden rule given by @HarshCasper - `"git pull origin once a day, keeps merge conflicts away"`

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by SangeetaMishra** [comment]
@atarax665  Can you please review the PR.

**Review by atarax665 — APPROVED** [review]

**Pull request merged**